### DISCUSSION TITLE
`x-transition.opacity` Sets `transform: scale(1)` Which Can Lead To Sideeffects

### DISCUSSION THREAD

**Discussion by fweth** [alpinejs/alpine#3129] [7. Bugs (archived)]
Hi, I just saw that when using `x-transition.opacity`, Alpine still sets a CSS transform with `scale(1)`, because it uses a [variable](https://github.com/alpinejs/alpine/blob/763e287842da1d9e0a32c52015cfd5c8f33dbac7/packages/alpinejs/src/directives/x-transition.js#L52) for the scale parameter.  I think that's a bit unfortunate, because every CSS transform creates a new stacking context (even when having no visible effect) which can lead to unexpected behaviour, e.g. [here](https://jsfiddle.net/1xb5v6eq/31/) the stacking order changes when `transform: scale(1);` is present.  Would be neat if `x-transition.opacity` really only sets `opacity` and `transition`!

Not sure if very elegant, but could write something like this:

```js

el._x_transition.leave.start = {
    opacity: 1,
    ...(wantsScale && {transform: `scale(1)`}),
}

el._x_transition.leave.end = {
    opacity: opacityValue,
    ...(wantsScale && {transform: `scale(${scaleValue})`}),
}
```

**Comment by SimoTod** [comment]
In your example, you are saying that the first div has a higher zIndex so it shows above the other one. 
Without transform, zIndex is ignored because the element is not positioned. 
It looks more like incomplete CSS rules to me (if you use z-index, you should position your element).

However, the suggested change should work and even if there are no real issues you are right, it probably shouldn't do anything that is not strictly required.

**Reply by fweth** [reply]
You are right, my example was not some good CSS, just wanted to point out that `transform: scale(1)` can make a difference.  [Here](https://jsfiddle.net/yog3xv51/14/) is another (imo better) one, if you want to place the `.aside` div between `.inner` and `.outer`, this is *only* possiible if `.outer` doesn't create a new stacking context (remove the line with `transform` to see what I mean), as far as I know, there is no way to circumvent this in CSS.
